### ISSUE TITLE
[P3][dogfood] UML sequence reports `truncated: true` on a complete 2-node / 1-edge direct call diagram

### ISSUE THREAD

**Issue by aimasteracc** [aimasteracc/tree-sitter-analyzer#787]
**Severity:** P3  |  **Surface:** UML sequence diagram (--uml sequence)
_Found by parallel dogfood audit round 2 on develop @ 471ac5f5; adversarially re-reproduced + dup-checked by an independent verifier._

### Observed
Envelope returns node_count=2, edge_count=1, nodes=['analyze_file','log_error'], a single edge analyze_file->log_error labelled 'call', verdict INFO — yet truncated: true. metadata shows path_count=28, so the flag is driven by the path-finder hitting its max_paths cap (call_path.py: truncated=len(paths)>=max_paths) propagated through uml_export.py:584 (truncated=result.truncated or len(first_hops)>max_hops), not by whether the rendered diagram omitted nodes/edges. The sequence diagram only renders paths[0]'s hops, so the rendered A->B relationship is fully present.

### Expected
`truncated` should reflect whether the emitted diagram omitted nodes/edges. A complete source->target sequence rendering should report truncated=false. If signalling that additional alternative call paths existed (path-finder hit max_paths), surface that under a separate field (e.g. paths_collapsed) rather than overloading `truncated`, which an agent reads as 'the diagram is incomplete'.

### Repro
```bash
cd /Users/aisheng.yu/git-private/tsa-dogfood && uv run python -m tree_sitter_analyzer --uml sequence --uml-source analyze_file --uml-target log_error
```

### Verifier notes
Reproduced the exact repro: `--uml sequence --uml-source analyze_file --uml-target log_error` returns node_count=2, edge_count=1, a single complete edge analyze_file->log_error labelled 'call', verdict INFO, metadata path_count=28 — yet truncated:true. Confirmed root cause in tree_sitter_analyzer/uml_export.py:584: `truncated=result.truncated or len(first_hops) > max_hops`. `result.truncated` originates in CallPathFinder (call_path.py:423/452/541/647) as `truncated=len(paths) >= max_paths`, i.e. it signals the path-FINDER hit its max_paths cap while enumerating alternative call paths. But the sequence diagram only renders paths[0]'s hops (uml_export.py:556-570), so the emitted A->B diagram is fully complete. The flag is therefore overloaded: it reports 'more alternative call paths existed in the search' but reads to an agent as 'the rendered diagram omitted nodes/edges'. Not a flag/user error — the command is valid and the diagram is correct; only the truncated semantics are misleading. P3: no wrong output, no crash; a metadata-semantics confusion an agent could misinterpret. Not a duplicate — #734 (call-path NO_PATH), #736/#737 (symbol-search/search caps), #755 (check-scale output size) all concern different surfaces; no open issue covers UML sequence truncated semantics.

**Comment by aimasteracc** [comment]
Fixed in a recent merge to develop. Verified as resolved.
